topologies/binding/binding.go

```diff
@@ -46,6 +47,13 @@ var _ = binding.Binding(&staticBind{})
 
 const resvID = "STATIC"
 
+var (
+	muIxWeb      sync.Mutex
+	muIxSession  sync.Mutex
+	ateIxWeb     = make(map[string]*ixweb.IxWeb)
```

Comment:
Let's make it not as a globally-scoped singleton, but a singleton that is tied to the staticBind instance. And I think you're right that each ATE will need its own session, so we will need a session map nonetheless.

(I guess there isn't a good name for per-instance singleton? Sorry for the confusion.)

Reply:
@greg-dennis if this is scoped to the staticBind instance, does it need to be protected by a mutex (i.e. thread-safe as opposed to thread-compatible)?

Reply:
> Let's make it not as a globally-scoped singleton, but a singleton that is tied to the staticBind instance. And I think you're right that each ATE will need its own session, so we will need a session map nonetheless.
> 
> (I guess there isn't a good name for per-instance singleton? Sorry for the confusion.)

do I need to change ateIxSession to be tied to staticBind instance as well?

Reply:
Yes, let's make them tied to staticBind as well.

Reply:
sure. thanks for quick confirmation. I will also make reserveATE/releaseATE/ixWebForATE/ixSessionForATE tied to staticBind.

Reply:
Hi @liulk, based on your suggestion, I made ateIxWeb, ateIxSession and related function tied to staticBind. I keep those mutex for now and will make changes based on discussion above. 
For singleton ateIxWeb/ateIxSession map, the current key is ate.Name which is same as ATE chassis IP. I think it would unique per ate for most of the most of cases in the binding file unless someone use different ate.Id with the same ATE chassis. What do you think about it? Do we need to change the key to be ate.Id?